src/pages/ClubDetail/ClubDetail.tsx

```diff
@@ -31,42 +48,93 @@ export const ClubDetail: React.FC = () => {
     }
   };
 
-  useEffect(() => {
-    const fetchClubDetail = async () => {
-      try {
-        const clubData = await getClubDetail(Number(id));
-
-        setClubDetail(clubData);
-      } catch (error) {
-        console.error('Error getting club detail:', error);
-      }
-    };
-
-    fetchClubDetail();
-  }, [id]);
+  const handleUpdate = async (updatedData: Partial<ClubType>) => {
+    try {
+      await updateClub(Number(id), updatedData);
+      const clubData = await getClubDetail(Number(id));
+      setClubDetail(clubData);
+    } catch (error) {
+      console.error('ClubDetail error:', error);
+    }
+  };
 
   return (
     <Layout>
-      <div>
-        {clubDetail ? (
-          <div className={style.clubDetailPage}>
-            <h1 className={style.title}>{clubDetail.name}</h1>
-            <p className={style.description}>{clubDetail.description}</p>
-            <p>members: </p>
-            {user?.id === String(clubDetail.ownerId) ? (
-              <button
-                type="button"
-                onClick={handleDelete}
-                className={style.deleteButton}
-              >
-                delete
-              </button>
-            ) : null}
+      {!clubDetail ? (
+        <Loading message="Loading book club details..." />
+      ) : (
```

Comment:
This is a good example for a early return. That is when you have a ternary that wraps basically the whole component